server/mobs.cpp

```diff
@@ -180,6 +181,28 @@ void Civilian::collisionResolution() {
 
 void Civilian::update()
 {
+	std::vector<MobManipulator> seenManips;
+	for (auto &m : gameState.mobManipulators) {
+		if (mobSeePoint(*this, m.pos, true))
+			seenManips.push_back(m);
+	}
+	if (seenManips.size() > 0) {
+		this->seenAManip = true;
+		auto last = seenManips.back();
+		if (!last.dispersor)
+			this->targetPosition = b2g(last.pos);
+		else {
+			auto toManip = this->body->GetPosition() - last.pos;
+			toManip.Normalize();
+			this->targetPosition = b2g(this->body->GetPosition() + toManip);
+		}
+		Mob::update();
+		return;
+	} else if (this->seenAManip) {
+		this->collisionResolution();
```

Comment:
This line makes many mobs disappear at random at the end of the mob manipulator effect duration.
1) It looks weird
2) If the tactical goal of MM is to simply expose players by moving every mob except for them, it's ok. But if I would like, for example, to create a dense mob cluster somewhere for a while, that's becoming impossible. If I understand the mechanics correctly, the mobs would go to the nearest waypoint anyway, so the cluster wouldn't be persistent.

Reply:
fixed